lib/excon/error.rb

```diff
@@ -50,7 +50,16 @@ class InvalidHeaderKey < Error; end
     class InvalidHeaderValue < Error; end
     class Timeout < Error; end
     class ResponseParse < Error; end
-    class ProxyConnectionError < Error; end
+
+    class ProxyConnectionError < Error
```

Comment:
Could we change this interface to look like the HTTPStatus errors? I don't think it makes sense to derive from that class, but I do think it probably makes sense to have `:response` as the attribute, and probably `:request` as well for consistency. Does that make sense? What do you think?

Here is the HTTPStatus code I'm referring to: https://github.com/excon/excon/blob/c39c6a2775ad3931a9ece2e847eb7d79a37ba048/lib/excon/error.rb#L67

Reply:
🤦‍♂ that makes perfect sense. We could also use some of the information in the original request. Appreciate the quick response!